### PR TITLE
Add ROS 2 Record & Replay and Fast DDS Spy descriptions

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

File: docs/rst/introduction/developer_tools.rst [hunk 1]
```diff
@@ -16,13 +16,17 @@ Developer Tools
...
+   tools/recordreplay.rst
+   tools/fastddsspy.rst
    tools/webots.rst
 
 - :ref:`ROS 2 QoS Profiles Manager <vulcanexus_qos_profiles_manager>`: tool suite to generate XML configuration files for Vulcanexus middleware Fast DDS.
 - :ref:`ROS 2 Monitor <vulcanexus_monitor>`: graphical desktop application to monitor *ROS 2* communications.
 - :ref:`Fast DDS Statistics Backend <fastdds_statistics_backend>`: *ROS 2 Monitor*'s backend. Can be leveraged with other monitoring frontend applications (e.g. :ref:`Prometheus <tutorials_tools_prometheus>`).
 - :ref:`ROS 2 Shapes Demo <vulcanexus_shapes_demo>`: first demo application to understand the most used *ROS 2* Quality of Service (QoS) and test DDS and *ROS 2* communication.
 - :ref:`ROS 2 Router <vulcanexus_router>`: end-user software application enabling the connection of distributed *ROS 2* environments.
+- :ref:`ROS 2 Record & Replay <vulcanexus_recordreplay>`: end-user software applications that efficiently saves messages published in a *ROS 2* environment and playback these messages in the order in which they were recorded.
```
Comment (jsantiago-eProsima) [inline docs/rst/introduction/developer_tools.rst:28]:
```suggestion
- :ref:`ROS 2 Record & Replay <vulcanexus_recordreplay>`: end-user software applications that efficiently save messages published in a *ROS 2* environment and playback these messages in the order in which they were recorded.
```

File: docs/rst/introduction/developer_tools.rst [hunk 2]
```diff
@@ -16,13 +16,17 @@ Developer Tools
...
+- :ref:`Fast DDS Spy <vulcanexus_fastddsspy>`: CLI interactive tool that allows to introspect a *ROS 2* environment in a human readable format.
```
Comment (jsantiago-eProsima) [inline docs/rst/introduction/developer_tools.rst:29]:
```suggestion
- :ref:`Fast DDS Spy <vulcanexus_fastddsspy>`: CLI interactive tool that allows the introspection of a *ROS 2* environment in a human readable format.
```

File: docs/rst/introduction/tools/fastddsspy.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
+.. _vulcanexus_fastddsspy:
+
+Fast DDS Spy
+============
+
+*Fast DDS Spy* is CLI interactive tool that allows to introspect a ROS 2 network in human readable format.
```
Comment (jsantiago-eProsima) [inline docs/rst/introduction/tools/fastddsspy.rst:6]:
```suggestion
*Fast DDS Spy* is CLI interactive tool that allows the introspection of a ROS 2 network in human readable format.
```

File: docs/rst/introduction/tools/fastddsspy.rst [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+It is possible to query the network about the Participants connected, their endpoints (Publishers and Subscriptions) and the topics they communicate in.
+It is also possible to see the user data sent through network topics in a schematic format in run time.
+
+Overview
+--------
+
+*Fast DDS Spy* is a tool that introspect or "sniffing" DDS packages in the network and maintain a local database that is accessible from a interactive CLI.
```
Comment (jsantiago-eProsima) [inline docs/rst/introduction/tools/fastddsspy.rst:13]:
```suggestion
*Fast DDS Spy* is a tool that introspects or "sniffs" DDS packages in the network and maintain a local database that is accessible from a interactive CLI.
```
Maybe "captures" is a better wording for what this does.

File: docs/rst/introduction/tools/fastddsspy.rst [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+*Fast DDS Spy* responds to user commands introduced by text and print in ``stdin`` the information requested.
```
Comment (jsantiago-eProsima) [inline docs/rst/introduction/tools/fastddsspy.rst:14]:
```suggestion
*Fast DDS Spy* responds to user commands introduced by text and prints to ``stdout`` the information requested.
```

File: docs/rst/introduction/tools/fastddsspy.rst [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+This tool has several commands to interact with, that allows to get information regarding the status of the network.
+It supports to list topics existing, list Participants, Publishers, Subscriptions and even read user data in real time in a human readable format.
```
Comment (jsantiago-eProsima) [inline docs/rst/introduction/tools/fastddsspy.rst:16]: I think support + gerund is the correct way of writing this, but I'm not 100% positive.

File: docs/rst/introduction/tools/fastddsspy.rst [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+.. figure:: /rst/figures/intro/tools/fastddsspy/fastddsspy.png
+    :align: center
+
+*Fast DDS Spy* is easily configurable and installed with a default setup, so that DDS topics, data types and entities are automatically discovered without the need to specify the types of data.
+This is because this tool exploits the DynamicTypes functionality of `eProsima Fast DDS <https://fast-dds.docs.eprosima.com>`_, the C++ implementation of the `DDS (Data Distribution Service) Specification <https://www.omg.org/spec/DDS/About-DDS/>`_ defined by the `Object Management Group (OMG) <https://www.omg.org/>`_.
```
Comment (jsantiago-eProsima) [inline docs/rst/introduction/tools/fastddsspy.rst:22]: `exploits` sounds so.. _negative_, almost as if it is not intended. `Leverages` might be better.

File: docs/rst/introduction/tools/recordreplay.rst [hunk 8]
```diff
@@ -0,0 +1,37 @@
+.. _vulcanexus_recordreplay:
+
+ROS 2 Record & Replay
+=====================
+
+*ROS 2 Record & Replay* (a.k.a. *DDS Record & Replay*) is an end-user software application that efficiently saves ROS 2 messages published into a DDS environment in a MCAP format database.
+Thus, the exact playback of the recorded network events is possible as the data is linked to the timestamp at which the original data was published.
+
+*ROS 2 Record & Replay* is easily configurable and installed with a default setup, so that DDS topics, data types and entities are automatically discovered without the need to specify the types of data recorded.
+This is because the recording tool exploits the DynamicTypes functionality of `eProsima Fast DDS <https://fast-dds.docs.eprosima.com>`_, the C++ implementation of the `DDS (Data Distribution Service) Specification <https://www.omg.org/spec/DDS/About-DDS/>`_ defined by the `Object Management Group (OMG) <https://www.omg.org/>`_.
```
Comment (jsantiago-eProsima) [inline docs/rst/introduction/tools/recordreplay.rst:10]: same comment regarding exploits

**Review by jsantiago-eProsima — CHANGES_REQUESTED** [review]

**Review by jsantiago-eProsima — APPROVED** [review]

**Comment by rsanchez15** [comment]
@Mergifyio backport humble

**Pull request merged**